### ISSUE TITLE
<MatrialUi.WithTheme> crash if classes property only is provided

### ISSUE THREAD

**Issue by unodgs** [jsiebern/bs-material-ui#17]
This code worked in previous versions:
```    
render: self => <MaterialUi.WithStyles
        classes=[{
            name: "fullScreen",
            styles: Style.make(~width="100vw", ~height="100vh", ())
        }, {
            name: "fullView",
            styles: Style.make(~width="100%", ~height="100%", ())
        }, {
```
but recently it stopped with following error:
![image](https://user-images.githubusercontent.com/7407/43355125-6d965988-9257-11e8-9b30-8a77c0a93c99.png)

If I add `classesWithTheme={theme => []}` then my code runs correctly again. But it shouldn't be necessary IMO.

**Comment by jsiebern** [comment]
You're absolutely right, that should work fine! I'll take a look!

**Comment by jsiebern** [comment]
Actually, that might be due to a regression in `bsb` that has been fixed since but not published to my knowledge! I'll have to test that.

**Comment by PTaylour** [comment]
> ✅ ~Think of a way to use theme => object pattern as withStyles argument(2017-11-25)~

Is this comment in the readme referring to the `classesWithTheme` prop?

If so, is this the only way to style with access to the theme, or is there also plans for it to be accessible via a ppx?

If there is anything I can do to help, give me a shout :)

**Comment by PTaylour** [comment]
I'll take a look :). Still new to all of this (and my bucklescript level of success has so far been mixed) — but no harm giving it a go!

**Comment by PTaylour** [comment]
@jsiebern So a good thing for me to start with would be to type the rest of the theme object here https://github.com/jsiebern/reason-mui-binding-generator/blob/master/src/fixed-modules/MaterialUi_Theme.re using `[@bs.deriving abstract]`?

**Comment by jsiebern** [comment]
@PTaylour Yes, I believe that would be a good place to start! I believe that bs still has no adequate way of converting / accessing multiple object levels, so that makes it a bit challenging.

Could you open a new issue for this please? I fixed the bug discussed in this issue just now and will close it once published

**Comment by jsiebern** [comment]
Could you kindly check `@next` and let me know if this issue can be closed?

**Comment by jsiebern** [comment]
Fixed in `0.4.2`
